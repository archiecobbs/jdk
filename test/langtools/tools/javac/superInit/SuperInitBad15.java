/*
 * @test /nodynamiccopyright/
 * @bug 8193760
 * @summary Verify this()/super() can only be invoked from constructors
 *
 * @compile/fail/ref=SuperInitBad15.out -XDrawDiagnostics SuperInitBad15.java
 */
class SuperInitBad15 {

    {
        super();
    }

    {
        this();
    }

    void normalMethod1() {
        super();
    }
    void normalMethod2() {
        this();
    }
    void normalMethod3() {
        Runnable r = () -> super();
    }
    void normalMethod4() {
        Runnable r = () -> this();
    }
}
