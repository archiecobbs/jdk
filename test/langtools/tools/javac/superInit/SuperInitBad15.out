SuperInitBad15.java:11:14: compiler.err.call.must.only.appear.in.ctor: super
SuperInitBad15.java:15:13: compiler.err.call.must.only.appear.in.ctor: this
SuperInitBad15.java:19:14: compiler.err.call.must.only.appear.in.ctor: super
SuperInitBad15.java:22:13: compiler.err.call.must.only.appear.in.ctor: this
SuperInitBad15.java:25:33: compiler.err.call.must.only.appear.in.ctor: super
SuperInitBad15.java:28:32: compiler.err.call.must.only.appear.in.ctor: this
6 errors
